### PR TITLE
Fixed terraform apply template step

### PR DESCRIPTION
**Pull request by mcasperson**
This is the same as https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/479, but without any tests.

### CODE REVIEW

File: octopusdeploy/schema_action_apply_terraform_template.go [hunk 1]
```diff
@@ -507,7 +518,7 @@ func flattenApplyTerraformTemplateAction(action *deployments.DeploymentAction) m
 				flattenedAction["aws_account"] = flattenTerraformTemplateAwsAccount(action.Properties)
 			}
 		case "Octopus.Action.Terraform.Template":
-			flattenedAction["template"] = v.Value
```
[the remark this answers is not in the document]
Comment (mcasperson) [reply octopusdeploy/schema_action_apply_terraform_template.go:510]: The original `template` parameter was a nested block of values that all related to terraform files in a package. And it had a fixed schema that required child properties defined in the `template` block.

If you look at https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/blob/9dfb9c3854206c737c061d5d663c81f5136475c5/octopusdeploy/schema_action_apply_terraform_template.go#L506 you can see that when the source is package, the provider maps the values in the `template` block to the individual Octopus properties that define which package, which file etc. So it was used (and is still used) so long as you used a package source for the step, not inline scripts.

When using inline scripts, the `Octopus.Action.Terraform.Template` value contains the terraform code. So the change here just exposes a second specialised property called `inline_template` that accepts the Terraform code, and has a schema that accepts a string.

So, in summary, `template` is still used when consuming code from a package, and `inline_template` now accepts a string for inline Terraform scripts. No one could ever create an inline terraform script, so there is no change for anyone who was successfully using the step.

**Review by zentron — APPROVED** [review]

**Review by hnrkndrssn — APPROVED** [review]

**Comment by mcasperson** [comment]
Fixes https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/pull/482
